src/main/scala/ai/privado/languageEngine/javascript/tagger/sink/RegularSinkTagger.scala

```diff
@@ -47,6 +48,40 @@ class RegularSinkTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelC
   override def runOnPart(builder: DiffGraphBuilder, ruleInfo: RuleInfo): Unit = {
     val sinks = cacheCall.methodFullName("(pkg.){0,1}(" + ruleInfo.combinedRulePattern + ").*").l
 
-    sinks.foreach(sink => addRuleTags(builder, sink, ruleInfo, ruleCache))
+    if (ruleInfo.id.equals(Constants.cookieWriteRuleId)) {
+      sinks.foreach(sink => {
+        val cookieNameArgument = sink.argument.or(_.argumentIndex(1), _.argumentName("name")).l
+        if (cookieNameArgument.nonEmpty) {
+          val cookieName = (cookieNameArgument.head match {
+            case node: Literal => node.code
+            case node: Identifier =>
+              cpg.assignment
```

Comment:
Just check if this logic holds all pattern coming:
`window.setCookie('username', 'afwef')`
`this.cookieService.set('Test', 'Hello World');`
`put(key, value, [options]);`
`putObject(key, value, [options])`
`document.cookie = 'default=%E5%8C%97'`